### PR TITLE
486 Test arbitrary user values valid

### LINKED ISSUE #486: Allow registration endpoint to accept arbitrary values that are saved to the user's profile
# Tasks
- [ ] Update docs

### PR DESCRIPTION
**Pull request by Mistweaver**
Test that optional values in the registration payload are assigned to the user object.  Also removes syn0.com domain usage from tests.

Closes #486

### CODE REVIEW

File: api/test/application/user/user-register-post.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+const factory = require("../../helpers/factory");
 const { expect } = Code;
-const uuid = require('uuid');
-const bookshelf = require('../../../src/lib/bookshelf');
-const querystring = require('querystring');
-const webhookService = require('../../../src/application/webhook/webhook-service');
+const uuid = require("uuid");
+const bookshelf = require("../../../src/lib/bookshelf");
+const querystring = require("querystring");
+const webhookService = require("../../../src/application/webhook/webhook-service");
```
Comment (Mistweaver) [inline api/test/application/user/user-register-post.js:14]: Just formatting, no imports were changed

File: api/test/application/user/user-register-post.js [hunk 2]
```diff
@@ -31,65 +32,127 @@ describe(`POST /api/invite`, () => {
...
-
-  it(`redirects and sets additional values to the user profile`, async () => {
+    client_id: client.get("client_id"),
+    response_type: "code id_token token",
+    scope: "openid profile app_metadata",
+    redirect_uri: client.related("redirect_uris").at(0).get("uri"),
+    nonce: "nonce",
+  });
+
+  it(`redirect as expected (status code 302)`, async () => {
```
Comment (Mistweaver) [inline api/test/application/user/user-register-post.js:45]: Modified the previous test to focus on just the redirect

File: api/test/application/user/user-register-post.js [hunk 3]
```diff
@@ -31,65 +32,127 @@ describe(`POST /api/invite`, () => {
...
+      .where({
+        email_lower: payload.email.toLowerCase(),
+      })
+      .fetch();
 
     expect(res.statusCode).to.equal(302);
-    expect(user.get('profile').name).to.equal(payload.name);
+  });
+
+  it(`accepts and sets all additional values to the user profile`, async () => {
```
Comment (Mistweaver) [inline api/test/application/user/user-register-post.js:70]: New test that analyzes the payload

File: api/test/application/user/user-register-post.js (hunk 1 above)
Comment (spruce-bruce) [reply api/test/application/user/user-register-post.js:14]: Is this a result of prettier running in your editor? If so can you turn that off in your editor for just this project OR just send a PR that prettiers the whole thing and adds prettier as a CI check?

**Review by spruce-bruce — APPROVED** [review]
Approved, with a request in one of my comments

**Pull request merged**